src/runtime_src/core/common/detail/linux/sysinfo.h

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+// Local - Include files
+#include "core/common/sysinfo.h"
+
+// 3rd Party Library - Include Files
+#include <boost/property_tree/ptree.hpp>
+#include <boost/format.hpp>
+#include <boost/property_tree/ini_parser.hpp>
+
+// System - Include Files
+#include <gnu/libc-version.h>
+#include <sys/utsname.h>
+#include <thread>
+
+#if defined(__aarch64__) || defined(__arm__) || defined(__mips__)
+  #define MACHINE_NODE_PATH "/proc/device-tree/model"
+#elif defined(__PPC64__)
+  #define MACHINE_NODE_PATH "/proc/device-tree/model-name"
+  // /proc/device-tree/system-id may be 000000
+  // /proc/device-tree/model may be 00000
+#elif defined (__x86_64__)
+  #define MACHINE_NODE_PATH "/sys/devices/virtual/dmi/id/product_name"
+#else
+#error "Unsupported platform"
+  #define MACHINE_NODE_PATH ""
+#endif
+
+namespace {
+static std::string machine_info()
+{
+  std::string model("unknown");
+  std::ifstream stream(MACHINE_NODE_PATH);
+  if (stream.good()) {
+    std::getline(stream, model);
+    stream.close();
+  }
+  return model;
+}
+static boost::property_tree::ptree
+glibc_info()
+{
+  boost::property_tree::ptree _pt;
+  _pt.put("name", "glibc");
+  _pt.put("version", gnu_get_libc_version());
+  return _pt;
+}
+
+} //end anonymous namespace
+
+
+namespace xrt_core::sysinfo {
+
+XRT_CORE_COMMON_EXPORT
+inline void
+get_os_info(boost::property_tree::ptree &pt)
+{
+  struct utsname sysinfo;
+  if (!uname(&sysinfo)) {
+    pt.put("sysname",   sysinfo.sysname);
+    pt.put("release",   sysinfo.release);
+    pt.put("version",   sysinfo.version);
+    pt.put("machine",   sysinfo.machine);
+  }
+
+  // The file is a requirement as per latest Linux standards
+  std::ifstream ifs("/etc/os-release");
+  if ( ifs.good() ) {
+      boost::property_tree::ptree opt;
+      boost::property_tree::ini_parser::read_ini(ifs, opt);
+      std::string val = opt.get<std::string>("PRETTY_NAME", "");
+      if (val.length()) {
+          if ((val.front() == '"') && (val.back() == '"')) {
+              val.erase(0, 1);
+              val.erase(val.size()-1);
+          }
+          pt.put("distribution", val);
+      }
+      ifs.close();
+  }
+
+  // BIOS info
+  std::string bios_vendor("unknown");
+  std::string bios_version("unknown");
+  std::ifstream bios_stream("/sys/class/dmi/id/bios_vendor");
+  if (bios_stream.is_open()) {
+    getline(bios_stream, bios_vendor);
+    pt.put("bios_vendor", bios_vendor);
+    bios_stream.close();
```

Comment:
close is not needed.  But if you want to close bios_stream before all the remaining code, the scope it.